src/main/java/dev/openfeature/sdk/ImmutableContext.java

```diff
@@ -79,14 +78,14 @@ public String getTargetingKey() {
     @Override
     public EvaluationContext merge(EvaluationContext overridingContext) {
         if (overridingContext == null || overridingContext.isEmpty()) {
-            return new ImmutableContext(this.asMap());
+            return new ImmutableContext(this.asUnmodifiableMap());
```

Comment:
If we're already in an `ImmutableContext` and there's nothing to add, why not just return `this`?

This goes in the direction @ssharaev suggested in https://github.com/open-feature/java-sdk/pull/1178/files#r1811813866

Reply:
Because then it's not a new instance, and I think always returning a new object from this method is less surprising than _sometimes_ returning the same one.